File: task_equivalents/llm_translations/gpt/optimized/python/data_manipulation_preprocessing/01_impute_encode_scale.py
import pandas as pd

auto = pd.read_csv("data/csv/Auto.csv", na_values="?", index_col=0).dropna()

auto["mpg01"] = (auto["mpg"] > auto["mpg"].median()).astype(int)
auto = auto.drop(columns=["mpg", "name"])

var_no_scale = ["cylinders", "year", "origin", "mpg01"]
var_to_scale = ["displacement", "horsepower", "weight", "acceleration"]

scaled_var = (auto[var_to_scale] - auto[var_to_scale].mean()) / auto[var_to_scale].std(ddof=1)
df = pd.concat([scaled_var, auto[var_no_scale]], axis=1)

x = df.iloc[:, :-1]
y = df["mpg01"]

File: task_equivalents/llm_translations/gpt/optimized/r/data_manipulation_preprocessing/01_impute_encode_scale.R
library(readr)
library(dplyr)

Auto <- read_csv("data/csv/Auto.csv", na = "?", show_col_types = FALSE) |>
tidyr::drop_na()

Auto <- Auto |>
mutate(mpg01 = if_else(mpg > median(mpg), 1L, 0L)) |>
select(-mpg, -name)

var_no_scale <- c("cylinders", "year", "origin", "mpg01")
var_to_scale <- c("displacement", "horsepower", "weight", "acceleration")

scaled_var <- scale(Auto[, var_to_scale])

df <- bind_cols(
as.data.frame(scaled_var),
Auto |> select(all_of(var_no_scale))
)

x <- df[, seq_len(ncol(df) - 1)]
y <- df$mpg01